src/main/java/bowling/FinalFrame.java

```diff
@@ -0,0 +1,67 @@
+package bowling;
+
+public class FinalFrame implements Frame {
+
+    private FinalThrown finalThrown;
+    private Pins bonusPins;
+
+    private FinalFrame(FinalThrown finalThrown) {
+        this.finalThrown = finalThrown;
+    }
+
+    public static FinalFrame of(int countOfPins) {
+        return new FinalFrame(new FinalThrown(Pins.from(countOfPins)));
+    }
+
+    public Frame bonusBowl(int countOfPins) {
+        bonusPins = Pins.from(countOfPins);
+        return this;
+    }
+
+    @Override
+    public boolean isFinished() {
+        if (bonusPins != null) {
+            return true;
+        }
+
+        if (secondPins() != 0 && getScore() != 10) {
+            return true;
+        }
+
+        return false;
```

Comment:
```suggestion
        return bonusPins != null || (secondPins() != 0 && getScore() != 10)
```

이런식으로 작성하면 좀 더 깔끔하게 의미 전달이 가능할수 있겟어요!